src/script/api/script_event_types.hpp

```diff
@@ -86,11 +88,18 @@ class ScriptEventVehicleCrashed : public ScriptEvent {
 	 */
 	SQInteger GetVictims() { return this->victims; }
 
+	/**
+	 * Get the CompanyID of the company owning the vehicle
+	 * @return The company owning the vehicle
+	 */
+	CompanyID GetVehicleOwner() { return this->company; }
+
 private:
 	TileIndex crash_site;     ///< The location of the crash.
 	VehicleID vehicle;        ///< The crashed vehicle.
 	CrashReason crash_reason; ///< The reason for crashing.
-	uint victims; ///< The number of victims.
+	uint victims;             ///< The number of victims.
+	CompanyID company;        ///< The company owning the vehicle.
```

Comment:
You want `ScriptCompany::CompanyID` here, and as return value of `GetVehicleOwner`. In the constructor you'll then need to call `ScriptCompany::ToScriptCompanyID(company)`.
Just take a peek at `ScriptEventPresidentRenamed` for how companies from the game engine need to be passed to the scripts.